### PR TITLE
fix(netops): Fix ip addr flush command

### LINKED ISSUE #5648: Typo in `distro.net_ops.flush_addr`
# Bug report
There seems to be a typo in `distro.net_ops.flush_addr`, which causes an invalid use of iproute2.
```
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/cloudinit/sources/helpers/digitalocean.py", line 81, in del_ipv4_link_local
    distro.net_ops.flush_addr(nic)
  File "/usr/lib/python3/dist-packages/cloudinit/net/netops/iproute2.py", line 144, in flush_addr
    subp.subp(["ip", "flush", "dev", interface])
  File "/usr/lib/python3/dist-packages/cloudinit/subp.py", line 295, in subp
    raise ProcessExecutionError(
cloudinit.subp.ProcessExecutionError: Unexpected error while running command.
Command: ['ip', 'flush', 'dev', 'eth0']
Exit code: 1
Reason: -
Stdout:
Stderr: Object "flush" is unknown, try "ip help".
```

This seems to be an obvious regression from https://github.com/canonical/cloud-init/commit/01027e5a0d056636ddde58b0c7e5cb537e3d43bc, since the previous use (https://github.com/canonical/cloud-init/commit/01027e5a0d056636ddde58b0c7e5cb537e3d43bc#diff-2fdeaaf3b1491e952eb4c6396294e92160441da230ba0a225eccab85ffe2f75dL89) had `addr`, while the new function (https://github.com/canonical/cloud-init/commit/01027e5a0d056636ddde58b0c7e5cb537e3d43bc#diff-f64049dab9ba0007ced723f722be7e0e70977baa9c4582161b9c04008ac6a4efR144) does not.

## Environment details
```
$ cloud-init --version
/usr/bin/cloud-init 24.2-0ubuntu1~24.04.2
```

### PR DESCRIPTION
**Pull request by holmanb**
```
fix(netops): Fix ip addr flush command

Fixes GH-5648
```

Fixes #5648

### CODE REVIEW

File: cloudinit/net/netops/iproute2.py [hunk 1]
```diff
@@ -141,4 +141,4 @@ def del_addr(interface: str, address: str):
 
     @staticmethod
     def flush_addr(interface: str):
-        subp.subp(["ip", "flush", "dev", interface])
+        subp.subp(["ip", "addr", "flush", "dev", interface])
```
Comment (aciba90) [inline cloudinit/net/netops/iproute2.py:144]: I recognize this method is very trivial to be unit-tested, but, do you think it is worth adding a broader unit test covering multiple `Iproute2`'s methods and covering this one?

**Review by aciba90 — APPROVED** [review]
LGTM, thanks!

**Review by a-dubs — APPROVED** [review]
thanks for adding all of those unit tests!

File: cloudinit/net/netops/iproute2.py (hunk 1 above)
Comment (aciba90) [reply cloudinit/net/netops/iproute2.py:143]: Many thanks!

**Review by aciba90 — APPROVED** [review]
Many thanks for adding unit-tests, @holmanb.

Feel free to merge, I am not doing it because I am not sure if you prefer squashing or not.

**Comment by holmanb** [comment]
Updated the commit messages, thanks for the review @aciba90!

**Pull request merged**